src/core/connection.c

```diff
@@ -2516,8 +2516,88 @@ QuicConnSetConfiguration(
             Status = QUIC_STATUS_INVALID_PARAMETER;
             goto Error;
         }
+        const uint8_t* AlpnList = Connection->Configuration->AlpnList;
```

Comment:
Let's move all this logic into a help function, perhaps in crypto.c.

Reply:
Done